### PR TITLE
Feature/GitHub actions workflow

### PR DESCRIPTION
**Pull request by snova-francescar**
Update the github action workflows to only apply to the Financial Assistant.
Add unit tests to the workflow.

### CODE REVIEW

**Review by snova-petrojm — APPROVED** [review]

File: financial_assistant/src/llm.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from langchain_core.output_parsers import JsonOutputParser
 from langchain_core.prompts import PromptTemplate
 from langchain_core.tools import StructuredTool, Tool
-from pydantic import BaseModel
```
Comment (snova-jorgep) [inline financial_assistant/src/llm.py:11]: consider changing this  for `from pydantic import BaseModel`
 keeping langchain pydantic_v1 will create conflict with LC0.3.1

Comment (snova-francescar) [reply financial_assistant/src/llm.py:11]: I will update all the `pydantic` dependencies when I upgrade my package to `langchain` 3.0.
Thank you for the reminder.

Comment (snova-francescar) [reply financial_assistant/src/llm.py:11]: I will open a PR with this upgrade soon.

Comment (snova-francescar) [reply financial_assistant/src/llm.py]: Now reverted to `pydantic`. Thank you!

**Pull request merged**